caas/kubernetes/provider/application/constraints.go

```diff
@@ -109,6 +113,51 @@ func ApplyConstraints(pod *core.PodSpec, appName string, cons constraints.Value,
 	return nil
 }
 
+// ApplyCharmConstraints applies the specified charm constraints to the pod.
+func ApplyCharmConstraints(pod *core.PodSpec, appName string,
+	charmContainerResourceRequirements caas.CharmContainerResourceRequirements) error {
+
+	if len(pod.Containers) == 0 {
+		return nil
+	}
+
+	requestValue := fmt.Sprintf("%dMi", charmContainerResourceRequirements.MemRequest)
+	limitValue := fmt.Sprintf("%dMi", charmContainerResourceRequirements.MemLimit)
```

Comment:
This allows us to set an invalid value in Kubernetes. If memory limit is zero then this is considered an error in Kubernetes. We should guard against this case and not set limit if that is the case. Probably the same for a request as well as a zero value request is the same as just not setting it.

I bring this up because this func is exported and we should not assume the caller has done what they think is correct considering we are translating from one modelling concept to another here.

Reply:
yup thanks for pointing this out tom! Not sure if a hard check for request is needed though, but perhaps let me put it in since it would be better to enforce callers to set a request than letting k8s default a req value.